sdk/core/core/src/az_auth.c

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_auth.h>
+
+#include <az_contract.h>
+#include <az_http_client.h>
+#include <az_http_request_builder.h>
+#include <az_json_read.h>
+#include <az_str.h>
+#include <az_uri.h>
+
+#include <assert.h>
+
+#include <_az_cfg.h>
+
+AZ_NODISCARD az_result az_auth_init_client_credentials(
```

Comment:
Please, confirm this name with @barcharcraz and @gilbertw. I'm already confused what name we should use. I prefer always use SVO (subject verb object) and the subject is the first parameter `self`. May be `out_self` for contructors.

Reply:
I'll check it in to master to unblock Victor, and will add another PR to follow up on this, as well as I'll improve URL encoding functions to work with span builder, so that string building code is going to take 4x less lines.